include/logger/common.hpp

```diff
@@ -22,5 +23,53 @@ struct ExtractedLogCredentials
   std::string function_name = "";
   int line_number = 0;
 };
+
+/**
+ * @brief Utiltiy function to check whether all the data like date, time etc are captured or not.
+ * @param log_ptr data to be tested
+ * @retval bool
+ */
+bool CheckValidLogCredentials(std::shared_ptr<logger::ExtractedLogCredentials> log_ptr)
+{
+  bool result = true;
+
+  // Checking for line number not be zero.
+  if (log_ptr->line_number == 0) {return false; }
+
+  // Checking for strings to be empty.
+  std::vector<std::string> test_string = { log_ptr->date, log_ptr->time, log_ptr->file_name, log_ptr->function_name };
+
+  for (auto str : test_string) {
+    if (str.empty()) {
+      return false;
+    }
+  }
+  return result;
+}
+
+/**
+ *@brief Utility function to convert log levels to string.
+ *@param level log level
+ *@retval string
+ */
+std::string LevelToString(logger::LEVEL level)
```

Comment:
As this function is not doing any input checks or changing anything, I suggest to change this  as const map.